src/TourManager/Repository/Repositories/Repository.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Linq.Expressions;
+using System.Collections.Generic;
+using Travely.TourManager.Service.Repositories;
+
+namespace Travely.TourManager.Repository.Repositories
+{
+    public class Repository<TEntity> : IRepository<TEntity> where TEntity : class
+    {
+        public ValueTask Add(TEntity entity)
```

Comment:
Why you are using ValueTask instead of Task ?

Reply:
ValueTask<T> is used to avoid allocating a new Task<T> when it's likely that the result of their operations will be available synchronously,

Add() and Find() usually completes synchronously

EF has changed its API, and if we return Task<T> in our repository, we mast wrap ValueTask<T> to Task<T> every time

https://github.com/dotnet/efcore/issues/15184